File: kangtaku/maximum_swap.cpp
class Solution {
public:
    int maximumSwap(int num) {
        string numStr = to_string(num);
        int n = numStr.size();
        int ans = num;
        for (int i = 0; i < n - 1; i++) {
            for (int j = i + 1; j < n; j++) {
                swap(numStr[i], numStr[j]);
                ans = max(ans, stoi(numStr));
                swap(numStr[i], numStr[j]);
            }
        }
        return ans;
    }
};

File: byssup/array/Maximum Swap.py
class Solution(object):
    def maximumSwap(self, num):
        """
        :type num: int
        :rtype: int
        """
        nn = [n for n in str(num)]
        for i, n in enumerate(nn):
            if not nn[i+1:]:
                continue

            max_val = max(nn[i+1:])
            if max_val > n:
                j = str(num).rfind(max_val)
                temp = nn[i]
                nn[i] = nn[j]
                nn[j] = temp
                return int(''.join(nn))
        return num